faststream/prometheus/middleware.py

```diff
@@ -0,0 +1,256 @@
+            return await call_next(msg)
+
+        messaging_system = self._settings_provider.messaging_system
+        consume_attrs = self._settings_provider.get_consume_attrs_from_message(msg)
+        destination_name = consume_attrs["destination_name"]
+
+        self._metrics.received_messages.labels(
+            broker=messaging_system,
+            handler=destination_name,
+        ).inc(consume_attrs["messages_count"])
+
+        self._metrics.received_messages_size.labels(
+            broker=messaging_system,
+            handler=destination_name,
+        ).observe(consume_attrs["message_size"])
+
+        err: Optional[Exception] = None
+
+        self._metrics.received_messages_in_process.labels(
+            broker=messaging_system,
+            handler=destination_name,
+        ).inc(consume_attrs["messages_count"])
+
+        start_time = time.perf_counter()
+
+        try:
+            result = await call_next(await self.on_consume(msg))
+
+        except Exception as e:
+            err = e
+            raise
+
+        finally:
+            duration = time.perf_counter() - start_time
+            self._metrics.received_messages_processing_time.labels(
+                broker=messaging_system,
+                handler=destination_name,
+            ).observe(duration)
+
+            self._metrics.received_messages_in_process.labels(
+                broker=messaging_system,
+                handler=destination_name,
+            ).dec(consume_attrs["messages_count"])
+
+            status = ProcessingStatus.acked
+
+            if msg.committed or err:
+                status = (
+                    PROCESSING_STATUS_BY_ACK_STATUS.get(msg.committed)  # type: ignore[arg-type]
+                    or PROCESSING_STATUS_BY_HANDLER_EXCEPTION_MAP.get(type(err))
+                    or ProcessingStatus.error
+                )
+
+            self._metrics.received_processed_messages.labels(
+                broker=messaging_system,
+                handler=destination_name,
+                status=status.value,
+            ).inc()
+
+            if status == ProcessingStatus.error:
+                self._metrics.messages_processing_exceptions.labels(
+                    broker=messaging_system,
+                    handler=destination_name,
+                    exception_type=type(err).__name__,
+                ).inc()
+
+        return result
+
+    async def publish_scope(
+        self,
+        call_next: "AsyncFunc",
+        msg: Any,
+        *args: Any,
+        **kwargs: Any,
+    ) -> Any:
+        if self._settings_provider is None:
+            return await call_next(msg, *args, **kwargs)
+
+        err: Optional[Exception] = None
+        start_time = time.perf_counter()
+
+        try:
+            result = await call_next(
+                await self.on_publish(msg, *args, **kwargs),
+                *args,
+                **kwargs,
+            )
+
+        except Exception as e:
+            err = e
```

Comment:
Why `_metrics.messages_publishing_exceptions` inc in finally block?

Reply:
fixed